app/uk/gov/hmrc/trackingconsentfrontend/assets/test/domain/userPreference.spec.ts

```diff
@@ -74,11 +73,53 @@ describe('User Preference Factory', () => {
     })
 
     it('should return undefined when no preferences set', () => {
-      testScope.cookieData.userConsent = undefined
+      setConsentCookie(undefined)
 
       expect(userPreferenceFactory().getPreferences()).toEqual(undefined)
     })
 
+    it('should return undefined when cookie value is not an object', () => {
+      setConsentCookie('abc')
+
+      expect(userPreferenceFactory().getPreferences()).toEqual(undefined)
+    })
+
+    it('should return undefined when cookie value is the wrong shape', () => {
+      setConsentCookie({ abc: 'abc' })
+
+      expect(userPreferenceFactory().getPreferences()).toEqual(undefined)
+    })
+
+    it('should only return partial data if partial data exists in the cookie', () => {
+      setConsentCookie({ preferences: { usage: true } })
+
+      expect(userPreferenceFactory().getPreferences()).toEqual({ usage: true })
+    })
+
+    it('should interpret boolean false correctly', () => {
+      setConsentCookie({ preferences: { usage: true, settings: false } })
+
+      expect(userPreferenceFactory().getPreferences()).toEqual({ usage: true, settings: false })
+    })
+
+    it('should ignore any non-boolean data', () => {
+      setConsentCookie({ preferences: { usage: true, settings: 'abc' } })
+
+      expect(userPreferenceFactory().getPreferences()).toEqual({ usage: true })
+    })
+
+    it('should ignore any null values', () => {
+      setConsentCookie({ preferences: { usage: true, settings: null } })
+
+      expect(userPreferenceFactory().getPreferences()).toEqual({ usage: true })
+    })
+
+    it('should ignore any undefined values', () => {
+      setConsentCookie({ preferences: { usage: true, settings: undefined } })
+
+      expect(userPreferenceFactory().getPreferences()).toEqual({ usage: true })
+    })
+
```

Comment:
Can we add a case for an unexpected version please?  `version: 2020-03-01` should be the only value we accept.